auth/sasl/authenticatee.go

```diff
@@ -0,0 +1,320 @@
+package sasl
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"code.google.com/p/gogoprotobuf/proto"
+	log "github.com/golang/glog"
+	"github.com/mesos/mesos-go/auth"
+	"github.com/mesos/mesos-go/auth/sasl/mech"
+	helper "github.com/mesos/mesos-go/auth/sasl/mesos"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesos/mesos-go/messenger"
+	"github.com/mesos/mesos-go/upid"
+	"golang.org/x/net/context"
+)
+
+var (
+	pidLock sync.Mutex
+	pid     int
+
+	UnexpectedAuthenticationMechanisms = errors.New("Unexpected authentication 'mechanisms' received")
+	UnexpectedAuthenticationStep       = errors.New("Unexpected authentication 'step' received")
+)
+
+type statusType int32
+
+const (
+	READY statusType = iota
```

Comment:
there's no sense exporting these since `statusType` is package private. consider renaming, for example, `READY` to something like `statusReady`